runtime/doc/lua.txt

```diff
@@ -2359,12 +2359,36 @@ find({names}, {opts})                                          *vim.fs.find()*
     The search can be narrowed to find only files or only directories by
     specifying {type} to be "file" or "directory", respectively.
 
+    Examples: >lua
+
+     -- location of Cargo.toml from the current buffer's path
+     local cargo = vim.fs.find('Cargo.toml', {
+       upward = true,
+       stop = vim.loop.os_homedir(),
+       path = vim.fs.dirname(vim.api.nvim_buf_get_name(0)),
+     })
+
+     -- list all test directories under the runtime directory
+     local test_dirs = vim.fs.find(
+       {'test', 'tst', 'testdir'},
+       {limit = math.huge, type = 'directory', path = './runtime/'}
+     )
+
+     -- get all the C++ source and header files inside lib/
```

Comment:
Nitpick: this assumes the reader knows what c++ files look like. Something like "get all files ending in .cpp and .hpp" or "get all files with extension .cpp and .hpp" would be preferable IMO